app/src/main/java/mozilla/lockbox/view/NetworkErrorHelper.kt

```diff
@@ -0,0 +1,106 @@
+        val toolbarMargin = view.resources.getInteger(R.integer.actionBarSize)
+        webparams.setMargins(0, toolbarMargin, 0, 0)
+        view.webView.layoutParams = webparams
+    }
+    // END FxALoginFragment
+
+    // START Item*Fragment
+    open fun showItemNetworkError(view: View) {
+        log.info("ELISE SHOW ERROR")
+        log.error(view.resources.getString(R.string.networkWarningMessage))
+        view.networkWarning.visibility = View.VISIBLE
+        view.warningMessage.text = view.resources.getString(R.string.no_internet_connection)
+    }
+
+    open fun hideItemListNetworkError(view: View) {
+        log.info(view.resources.getString(R.string.network_connection_success))
+        view.networkWarning.visibility = View.GONE
+        view.networkWarning.layoutParams.height = R.dimen.hiddenNetworkError
+
+        // set margin of entriesView to show only toolbar
+        val marginLayoutParams =  ViewGroup.MarginLayoutParams(view.refreshContainer.entriesView.layoutParams)
+        marginLayoutParams.leftMargin = 0
+        marginLayoutParams.topMargin = 0
+        marginLayoutParams.rightMargin = 0
+        marginLayoutParams.bottomMargin = 0
+
+        view.refreshContainer.entriesView.layoutParams = marginLayoutParams
+    }
+
+    open fun hideItemDetailNetworkError(view: View) {
+        log.info(view.resources.getString(R.string.network_connection_success))
+        view.networkWarning.visibility = View.GONE
+        view.networkWarning.layoutParams.height = R.dimen.hiddenNetworkError
+
+        // set margin of card_view to show only toolbar
+        val viewParams = view.card_view.layoutParams as ConstraintLayout.LayoutParams
+        viewParams.setMargins(0, 0, 0, 0)
+        view.card_view.layoutParams = viewParams
+    }
+    // END Item*Fragment
+
+    // START AppWebPageFragment
+    open fun showWebPageNetworkError(view: View) {
+        log.error(view.resources.getString(R.string.networkWarningMessage))
+        view.networkWarning.visibility = View.VISIBLE
+
+        // set margin of webview to show warning message and toolbar
+        val webParams = view.webView.layoutParams as ConstraintLayout.LayoutParams
+        val toolbarPlusWarningMargin = view.resources.getInteger(R.integer.network_error)
+        webParams.setMargins(0, toolbarPlusWarningMargin, 0, 0)
+        view.webView.layoutParams = webParams
+
+        view.warningMessage.text = view.resources.getString(R.string.no_internet_connection)
+    }
+
+    open fun hideWebPageNetworkError(view: View) {
+        log.info("Network successfully reconnected.")
+        view.networkWarning.visibility = View.VISIBLE
+
+        // set margin of webview to show only toolbar
+        val webparams = view.webView.layoutParams as ConstraintLayout.LayoutParams
+        webparams.setMargins(0, 0, 0, 0)
+        view.webView.layoutParams = webparams
+    }
+    // END AppWebPageFragment
+}
```

Comment:
There's lots of duplicated code in this file. Can we reduce this down, perhaps by making the XML more uniform. e.g. make `ConstraintLayout` the default layout we use everywhere.

You can also pass in parameters to the show and hide methods e.g. 

```
open fun hideNetworkError(networkWarning: View, squashedView: View) {
   networkWarning.visibility = View.GONE
   val layoutParams = squashedView.layoutParams
   if (layoutParams is ConstraintLayout.LayoutParams) {
        layoutParams.setMargins(0,0,0,0)
   }
   squashedView.layoutParams = layoutParams
}
```

Thirdly, you could just put the snackbar over the view, not doing the harder job of fitting it in between the squashed view and either the toolbar or the top of the screen.

Reply:
I worked for a while on making the layouts more uniform, but found that it would take a lot of time to reformat every fragment that way, so I abandoned that in favor of this not-so-nice formatting.

I'd like to talk about the 3rd point with you (I don't think I'm sure about how that would fit into the designs).